components/ucd/category/src/lib.rs

```diff
@@ -0,0 +1,313 @@
+// Copyright 2012-2015 The Rust Project Developers.
+// Copyright 2017 The UNIC Project Developers.
+//
+// See the COPYRIGHT file at the top-level directory of this distribution.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+
+#![deny(unsafe_code, missing_docs)]
+
+//! # UNIC — UCD — Category
+//!
+//! A component of [`unic`: Unicode and Internationalization Crates for Rust](/unic/).
+//!
+//! Unicode General Category.
+//!
+//!
+
+extern crate unic_ucd_core;
+
+use std::cmp::Ordering;
+
+use unic_ucd_core::UnicodeVersion;
+
+
+/// The [Unicode version](http://www.unicode.org/versions/) of data
+pub const UNICODE_VERSION: UnicodeVersion = include!("tables/unicode_version.rsv");
+
+/// Represents the Unicode Character
+/// [*General Category*](http://unicode.org/reports/tr44/#General_Category) property.
+///
+/// * <http://unicode.org/reports/tr44/#General_Category_Values>
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub enum GeneralCategory {
```

Comment:
I recently started putting all type-level abstraction in their own module. So, `Age` enum and its various implementations and tests are in `age.rs`. And, `char`/`str` helper traits go into `traits.rs`, and only a couple of needed things get `pub use`d here in `lib.rs` from different modules.

It has shown to be a good way to keep unrelated matters separate. Would be great to do the same here.

PS. I'll start writing these in `CONTRIBUTING.md`. (https://github.com/behnam/rust-unic/issues/36)